### DISCUSSION TITLE
Frame security always returns verified = false

### DISCUSSION THREAD

**Discussion by chilaraiSxt** [wevm/frog-archived#462] [Q&A] [answered]
I am unable to get a valid `fid` from the frames as `verified` always returns `false`. How do I fix that?
Here is my partial code

```ts
import { Button, Frog } from "frog";
import { devtools } from "frog/dev";
import { neynar } from "frog/hubs";
import { handle } from "frog/next";
import { serveStatic } from "frog/serve-static";

const app = new Frog({
    assetsPath: "/",
    basePath: "/api",
    hub: neynar({ apiKey: "HUB_KEY_DUMMY" }),
    title: "XYS",
    secret: "SOME_SECERT_FROM_ENV",
});


app.frame("/", (c) => {
    const { buttonValue, status, verified, frameData } = c;

    if (!verified) {
        return c.res({
            headers: { contentType: "image/png" },
            image: invalidLogin(),
        });
    } else {
        const { fid } = frameData || {};
        .....
        return c.res({
             ....
        })
     }
})

devtools(app, { serveStatic });

export const GET = handle(app);
export const POST = handle(app);
``` 

I even tried it on real Warpcast, but still get a `verified = false`

**Comment by chilaraiSxt** [comment, marked as answer]
Seems like the frame validation and other stuffs doesn't work on the landing frame. Only after some action is taken, the next frame gets all the information.

**Reply by dalechyn** [reply]
sorry I missed the notification of this post!

what you're saying makes total sense. since there is no POST payload to verify, it might be that the library is assuming that a frame is not verified.

do you think having this `true` for the first frame would be something you would expect?
